examples/08_join_aggregation.py

```diff
@@ -1,302 +1,548 @@
-]
-
-cv = GridSearchCV(pipeline, param_grid, cv=TimeSeriesSplit(n_splits=10))
-cv.fit(X, y)
-
-results = pd.DataFrame(cv.cv_results_)
-
-cols = [f"split{idx}_test_score" for idx in range(10)]
-results = results.set_index("param_aggtarget-2__operation")[cols].T
-results
-
-###############################################################################
-# The score used in this regression task is the R2. Remember that the R2
-# evaluates the relative performance compared to the naive baseline consisting
-# in always predicting the mean value of ``y_test``.
-# Therefore, the R2 is 0 when ``y_pred = y_true.mean()`` and is upper bounded
-# to 1 when ``y_pred = y_true``.
+# ``goods_code`` is slightly different, as it represents some merchant IDs, which
+# co-occurs for different products. Therefore, we encode it with a |TargetEncoder| as
+# we previously did.
 #
-# To get a better sense of the learning performances of our simple pipeline,
-# we also compute the average rating of each movie in the training set,
-# and uses this average to predict the ratings in the test set.
-from sklearn.metrics import r2_score
+# To later perform the joiner operation, we must keep the basket ``ID`` with
+# ``"passthrough"``.
+from skrub import MinHashEncoder
 
 
-def baseline_r2(X, y, train_idx, test_idx):
-    """Compute the average rating for all movies in the train set,
-    and map these averages to the test set as a prediction.
+def get_X_y(data):
+    return data.drop(columns=[target_col]), data[target_col]
 
-    If a movie in the test set is not present in the training set,
-    we simply predict the global average rating of the training set.
-    """
-    X_train, y_train = X.iloc[train_idx].copy(), y.iloc[train_idx]
-    X_test, y_test = X.iloc[test_idx], y.iloc[test_idx]
 
-    X_train["y"] = y_train
+tic = time()
+vectorizer = TableVectorizer(
+    high_cardinality=MinHashEncoder(),  # applied on ["item", "model", "make"]
+    specific_transformers=[
+        (TargetEncoder(), ["goods_code"]),
+        ("passthrough", ["ID"]),
+    ],
+)
 
-    movie_avg_rating = X_train.groupby("movieId")["y"].mean().to_frame().reset_index()
+products_transformed = vectorizer.fit_transform(*get_X_y(products))
+time_to_fit = time() - tic
 
-    y_pred = X_test.merge(movie_avg_rating, on="movieId", how="left")["y"]
-    y_pred = y_pred.fillna(y_pred.mean())
+TableReport(products_transformed)
 
-    return r2_score(y_true=y_test, y_pred=y_pred)
+# %%
+# Let's now detail how to leverage |AggJoiner| here. We have just encoded each product
+# attributes, and now we need to somehow aggregate these product encodings into their
+# respective baskets.
+#
+# By aggregating instead of concatenating, we obtain an invariant number of columns,
+# and we remove the sparsity of the dataset.
+#
+# But which aggregation operation should we choose? Since the |MinHashEncoder| hashes
+# ngrams with different hashing functions and return their minimum, it makes sense to
```

Comment:
this part about the minhash is probably hard to follow for someone who isn't very familiar with the minhash and skrub jargon